R/update-regional.R

```diff
@@ -0,0 +1,82 @@
+require(EpiNow2, quietly = TRUE)
+require(covidregionaldata, quietly = TRUE)
+require(data.table, quietly = TRUE)
+require(future, quietly = TRUE)
+require(lubridate, quietly = TRUE)
+
+# Load utils --------------------------------------------------------------
+
+source(here::here("R", "utils.R"))
+
+
+#' update.regional
+#'
+#' @description Processes regional data in an abstract fashion to reduce code duplication
+#' @param region_name String, name of region, used for filepaths
+#' @param region_identifier String, region identifier used by covidregionaldata to get the data #todo audit which countries this differs from name and standardise
+#' @param case_modifier_function Function, passed the cases, should return the cases. Method of modifying the returned data for a specific region if needed
+#' @param generation_time optional overrides for the loaded rds file. If present won't be reloaded from disk.
+#' @param incubation_period optional overrides for the loaded rds file. If present won't be reloaded from disk.
+#' @param reporting_delay optional overrides for the loaded rds file. If present won't be reloaded from disk.
+#' @param cases_region_source string, optional specification of where to get the list of regions from the cases dataset
+#' @param region_scale string, specify the region scale to epinow
+update.regional <- function(region_name, region_identifier, case_modifier_function = NULL, generation_time = NULL, incubation_period = NULL, reporting_delay = NULL, cases_region_source = NULL, region_scale = NULL) {
+  futile.logger::flog.info("Processing regional dataset for %s", region_name)
+  # setting debug level to trace whilst still in beta. #ToDo: remove this line once production ready
+  futile.logger::flog.threshold(futile.logger::TRACE)
+
+  # Update delays -----------------------------------------------------------
+
+  if (is.null(generation_time)) {
+    generation_time <- readRDS(here::here("data", "generation_time.rds"))
+  }
+  if (is.null(incubation_period)) {
+    incubation_period <- readRDS(here::here("data", "incubation_period.rds"))
+  }
+  if (is.null(reporting_delay)) {
+    reporting_delay <- readRDS(here::here("data", "onset_to_admission_delay.rds"))
+  }
+
+  # Get cases  ---------------------------------------------------------------
+
+  futile.logger::flog.trace("Getting regional data")
+  cases <- data.table::setDT(covidregionaldata::get_regional_data(country = region_identifier))
+  if (typeof(case_modifier_function) == "closure") {
+    futile.logger::flog.trace("Modifying regional data")
+    cases <- case_modifier_function(cases)
+  }
+  if (is.null(cases_region_source)) {
+    futile.logger::flog.trace("Cleaning regional data")
+    cases <- clean_regional_data(cases)
+  }else {
+    futile.logger::flog.trace("Cleaning regional data with %s as region source", cases_region_source)
+    cases <- clean_regional_data(cases[, region := eval(parse(text = cases_region_source))])
```

Comment:
This is a nice way of doing this. We are looking at adding a switch in covidregionaldata so that the regional data comes out in the format we need rather than having to reset for each region.

Reply:
That would be tidier - give a consistent structure to the data so that every data source obeys the same interface

Reply:
yup - agreed. We originally had this but ended up moving away from it as other users (i.e in-country) wanted familiar region names. I'll wait until this PR goes in and the update `covidregionaldata` and make the change here.